msys2: uninstall pkgconf

ericLemanissier opened the pull request:
pkgconf  should be provided by conan
also, hide the msys2 library folder. This is useful to avoid accidentally linking to a msys2 library instead of one from a conan dependency.
also also, make it clear that msys2 uses rolling releases

Specify library name and version:  **msys2/***

this would also solve #5322 failure

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

ericLemanissier commented on recipes/msys2/all/conanfile.py:
```diff
@@ -208,6 +208,8 @@ def _do_build(self):
         with tools.chdir(os.path.join(self._msys_dir, "usr", "bin")):
             for package in packages:
                 self.run('bash -l -c "pacman -S %s --noconfirm"' % package)
+            for package in ['pkgconf']:
+                self.run('bash -l -c "pacman -Rs $(pacman -Qsq %s) --noconfirm"' % package)
```
```suggestion
            for package in ['pkgconf', 'autoconf', 'automake', 'gettext-devel', 'libiconv-devel', 'libtool', 'make']:
                self.run('bash -l -c "pacman -Rs $(pacman -Qsq ^%s) --noconfirm"' % package)
```

these are other packages provided by conan. Do you have objections to removing these too ?

SpaceIm replied:
No objection, but maybe list & fix all CCI recipes silently relying on autotools & pkgconf before merging this PR.

madebr replied:
What's the content of `/usr/lib(64)/pkgconfig`?
Perhaps this folder can be removed too?

ericLemanissier replied:
```
$ ls /usr/lib/pkgconfig/
autoopts.pc  libmagic.pc          python-3.8.pc     python3.pc
libalpm.pc   python-3.8-embed.pc  python3-embed.pc  tcl.pc
```
```
$ ls /usr/share/pkgconfig/
bash-completion.pc  libmakepkg.pc
```

madebr replied:
Let's keep the folder and only remove it (or its contents) only when it is starting to cause problems.

SpaceIm replied:
Removing `make` from msys2 could break MiNGW builds of many recipes I think.

ericLemanissier replied:
To sum this up, I think this is too big of a change for now. (removing pkgconf should be safe, because It was not [present in precedent versions of msys](https://github.com/msys2/MSYS2-packages/commit/3009e7e836cc46b0ff4a9832f49741c66d1cfff7#diff-ca27ad8b0bbc2ac3072fcc97ec32b241e461331d0baef59595a03cff6ad6f013))

madebr commented:
@ericLemanissier 
To avoid too many rebuilds, maybe you can include the simple change from #5244 into this pr as well?

ericLemanissier commented:
Does anyone have an idea about this CI error ?
```
ERROR: C:\J\w\BuildSingleReference@8\.conan\data\msys2\cci.latest\_\_\package\eee3fba89db6d777329de604625af8c30d46f080

Folder: Cannot change permissions for C:\J\w\BuildSingleReference@8/s\eb6155\1\bin\msys64\usr\bin\dirmngr.exe! Exception info: (<class 'PermissionError'>, PermissionError(13, 'Access is denied'), <traceback object at 0x0000027A5D603188>)
Couldn't remove folder, might be busy or open
Close any app using it, and retry
```
It should not be related to locks, because there is only one configuration to build with this PR

ericLemanissier commented:
@jgsogo @danimtb @uilianries I really don't understand this error. I don't even understand at which step of the process it happens. Would you have a pointer for me to look at ?

SSE4 commented:
seems like there is `dirmgr.exe` running on CI, and msys2 build directory in the local cache, which needs to be removed before package is built. it fails long before the recipe code is executed, which makes it problematic to handle.

madebr reviewed (approved):
c3i succeeded!
Everybody approve before it changes its mind! 
:rofl:

ericLemanissier commented:
I think I understood and fixed in a reliable way the failure which happened above:

1. the build failed because of workarround: fixed by https://github.com/conan-io/conan-center-index/pull/5324/commits/dd7429057afbb4166cfa4b130cc15a108866b47a
2. after a build failure, some processes were kept alive, which led to the failure to remove dirmngr.exe: fixed by https://github.com/conan-io/conan-center-index/pull/5324/commits/e01df91afad71e731b22cf56b8f8028783f88579 which made

SpaceIm commented on recipes/msys2/all/conandata.yml:
```diff
@@ -1,34 +1,7 @@
-                 "https://sourceforge.net/projects/msys2/files/Base/x86_64/msys2-base-x86_64-20210105.tar.xz",
-          ]
-          sha256: "982e54de087d53adfc6a8caf7614d4a7add36dd02dcb0b7838060dd893e9f596"
+  "cci.latest":
```
what is the consequences for CCI recipes depending on removed versions?

ericLemanissier replied:
Existing binaries will remain available, and their hashes should not change because this recipe does not have requirements.
see also https://github.com/conan-io/conan-center-index/pull/4286#issuecomment-796737851

SpaceIm commented on recipes/msys2/all/conanfile.py:
```diff
@@ -60,81 +60,14 @@ class MSYS2Conan(ConanFile):
         if self.settings.os != "Windows":
             raise ConanInvalidConfiguration("Only Windows supported")
-        if tools.Version(self.version) >= "20210105" and self.settings.arch != "x86_64":
+        if self.settings.arch != "x86_64":
             raise ConanInvalidConfiguration("Only Windows x64 supported")
```
It means that CCI recipes depending on msys2 won't support anymore building on 32 bits architecture.

ericLemanissier replied:
yes, but msys2/20210105 did not support 32bits either

SpaceIm replied:
this is why I did not update msys2 in CCI recipes requiring it.

ericLemanissier replied:
ok, so users requiring 32bits can use msys2/20200517. What is the problem/what do you suggest ?

SpaceIm replied:
build requirements can't be overridden. The problem is that we can't update msys2 version in CCI recipes depending on it, or we accept to explicitly drop support of 32 bits build machine in these recipes.

ericLemanissier replied:
You can use another msys2 installation by setting CONAN_BASH_PATH env var. Also, is it not possible to use cross-compilation profiles from x86_64 to x86 in this case, and use 64bits msys2 to compile 32bits libraries ?
